src/praisonai-agents/praisonaiagents/memory/memory.py

```
@@ -693,6 +693,16 @@ def search_short_term(
                 return []
         
         else:
+            # Delegate to the active adapter when provider falls back to
+            # sqlite/in_memory. This avoids the legacy short_mem schema mismatch
+            # where the adapter creates short_term_memory but legacy queries
+            # short_mem (which is never created by the adapter).
+            if getattr(self, "provider", None) in ("sqlite", "in_memory") and getattr(self, "memory_adapter", None):
+                try:
+                    return self.memory_adapter.search_short_term(query, limit=limit, **kwargs)
+                except Exception as e:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)